tests/explicit-nulls/neg/i17467.scala

```diff
@@ -0,0 +1,16 @@
+object Test:
+  def test(): Unit =
+    val a1: String = "foo"
+    val a2: a1.type = null // error
+
+    val b1: String | Null = "foo"
+    val b2: b1.type = null // error
+    summon[Null <:< b1.type] // error
+
+    /* The following would be sound, but it would require a specific subtyping
+     * rule (and implementation code) for debatable value. So it is an error.
+     */
+    val c1: Null = null
+    val c2: h1.type = null // error
```

Comment:
Here's the problem with making checkfiles optional... what's the error?  Are we meant to be testing `h1` wasn't found?

Reply:
Oh my god :man_facepalming: 
Fixed, and added check files.